### DISCUSSION TITLE
MQTT Java client fails to connect when EMQX MySQL authentication is enabled

### DISCUSSION THREAD

**Discussion by neilyoung** [emqx/emqx#12538] [General]
I'm having enabled EMQX MySQL authentication. That works for itself. It is running on a different database with another user but on the same MySQL instance, a Java service is using. When the authentication is disabled in the EMQX dashboard and the Java server starts, all is fine. If authentication is enabled when the Java server starts, the Java server is `unable to connect or subscribe`.

The error which pops up every now and then in Java server's log is:

```
2024-02-19 17:10:34.110 ERROR 71480 --- [           main] .m.i.MqttPahoMessageDrivenChannelAdapter : Error connecting or subscribing to [sys/product/+/status, thing/product/+/requests]

org.eclipse.paho.client.mqttv3.MqttSecurityException: Not authorized to connect
	at org.eclipse.paho.client.mqttv3.internal.ExceptionHelper.createMqttException(ExceptionHelper.java:28) ~[org.eclipse.paho.client.mqttv3-1.2.5.jar!/:na]
	at org.eclipse.paho.client.mqttv3.internal.ClientState.notifyReceivedAck(ClientState.java:1053) ~[org.eclipse.paho.client.mqttv3-1.2.5.jar!/:na]
	at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:151) ~[org.eclipse.paho.client.mqttv3-1.2.5.jar!/:na]
	at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]

```

So in other words: The EMQX connection seems to block my MQTT client connection and I have no clue, why.

How could I narrow down the problem?

**Comment by neilyoung** [comment]
Disregard, I'm just too stupid today :)
